File: Chapter04/FS/ColorLoop/ColorLoop/ColorLoop/ColorLoopPage.fs
namespace ColorLoop

open Xamarin.Forms

type ColorLoopPage() = 
    inherit ContentPage()

    let colors = 
        [
            (Color.White, "White")
            (Color.Silver, "Silver")
            (Color.Gray, "Gray")
            (Color.Black, "Black")
            (Color.Red, "Red")
            (Color.Maroon, "Maroon")
            (Color.Yellow, "Yellow")
            (Color.Olive, "Olive")
            (Color.Lime, "Lime")
            (Color.Green, "Green")
            (Color.Aqua, "Aqua")
            (Color.Teal, "Teal")
            (Color.Blue, "Blue")
            (Color.Navy, "Navy")
            (Color.Pink, "Pink")
            (Color.Fuchsia, "Fuchsia")
            (Color.Purple, "Purple")
        ]

    let stackLayout = StackLayout()

    do for color in colors
        do stackLayout.Children.Add(Label(Text = snd color,
                                          TextColor = fst color,
                                          FontSize = Device.GetNamedSize(NamedSize.Large, typeof<Label>)))

    do base.Padding <- match Device.RuntimePlatform with 
                            | Device.iOS -> Thickness(5.0, 20.0, 5.0, 5.0)
                            | _ -> Thickness(5.0, 5.0, 5.0, 5.0)

    do base.Content <- stackLayout
    

File: Chapter04/ColorLoop/ColorLoop/ColorLoop/ColorLoopPage.cs
using System;
using Xamarin.Forms;

namespace ColorLoop
{
    class ColorLoopPage : ContentPage
    {
        public ColorLoopPage()
        {
            var colors = new[] 
            {
                new { value = Color.White, name = "White" },
                new { value = Color.Silver, name = "Silver" },
                new { value = Color.Gray, name = "Gray" },
                new { value = Color.Black, name = "Black" },
                new { value = Color.Red, name = "Red" },
                new { value = Color.Maroon, name = "Maroon" },
                new { value = Color.Yellow, name = "Yellow" },
                new { value = Color.Olive, name = "Olive" },
                new { value = Color.Lime, name = "Lime" },
                new { value = Color.Green, name = "Green" },
                new { value = Color.Aqua, name = "Aqua" },
                new { value = Color.Teal, name = "Teal" },
                new { value = Color.Blue, name = "Blue" },
                new { value = Color.Navy, name = "Navy" },
                new { value = Color.Pink, name = "Pink" },
                new { value = Color.Fuchsia, name = "Fuchsia" },
                new { value = Color.Purple, name = "Purple" }
            };

            StackLayout stackLayout = new StackLayout();

            foreach (var color in colors)
            {
                stackLayout.Children.Add(
                    new Label
                    {
                        Text = color.name,
                        TextColor = color.value,
                        FontSize = Device.GetNamedSize(NamedSize.Large, typeof(Label))
                    });
            }

            Padding = new Thickness(5, Device.RuntimePlatform == Device.iOS ? 20 : 5, 5, 5);
            Content = stackLayout;
        }
    }
}
